### PR TITLE
CGE-99 (personal area)

### PR DESCRIPTION
**Pull request by Dariy98**

### CODE REVIEW

File: src/main/resources/react/src/components/Card/inboxMessageSvg.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	return(
+		<InboxDiv>
+			<InboxSvg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
+				<path
+					d="M40.9567 7.03256C36.4202 2.49641 30.402 0 23.9879 0C17.5739 0 11.5557 2.49641 7.01914 7.03256C-1.74942 15.8004 -2.35835 29.8351 5.50697 39.3032C4.73566 40.8558 3.49751 42.6622 1.66057 43.5653C0.777627 44.0017 0.280336 44.9455 0.432568 45.9197C0.5848 46.8939 1.33581 47.655 2.3101 47.8072C2.76679 47.8782 3.41632 47.9493 4.20792 47.9493C6.32902 47.9493 9.45485 47.452 12.6517 45.1484C16.214 47.0664 20.1111 48 23.9778 48C30.1889 48 36.339 45.5949 40.9669 40.9674C45.5034 36.4313 48 30.4135 48 24C48 17.5865 45.4932 11.5687 40.9567 7.03256ZM39.0183 39.0292C32.1679 45.8791 21.5624 47.2389 13.2201 42.3374C12.672 42.0127 11.9921 42.1142 11.5557 42.5505C11.5151 42.5708 11.4745 42.6013 11.444 42.6317C8.69369 44.7628 5.99411 45.2093 4.20792 45.2093H4.19777C6.25798 43.7074 7.55703 41.4748 8.31819 39.7395C8.43997 39.4452 8.46027 39.1408 8.38923 38.8567C8.35878 38.5827 8.24714 38.3087 8.05432 38.0854C0.625395 29.693 1.01105 16.9066 8.94741 8.97082C17.239 0.679915 30.7267 0.679915 39.0082 8.97082C47.3099 17.2617 47.3099 30.7383 39.0183 39.0292Z"
```
Comment (rpkyrych) [inline src/main/resources/react/src/components/Card/inboxMessageSvg.js:9]: If I understand right it's an image. Is it possible to extract all such images into a file with detailed information about their relations?

File: src/main/resources/react/src/components/title_Big_Blue/title_Big_Blue.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
 import React from "react";
-import "./title.scss";
+import styled from 'styled-components';
 
-const TitleBigBlue = ({whatClass, text}) => {
+const H2 = styled.h2`
```
Comment (rpkyrych) [inline src/main/resources/react/src/components/title_Big_Blue/title_Big_Blue.js:4]: Why didn't you extract it into .css file?

File: src/main/resources/react/src/pages/UserInfo/components/buttonsAddRemoveChild/blockButtons.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	return (
+		<ButtonsBlock mb={ mb }>
+			<AddRemoveDiv>
+				<ButtonsAddRemoveChild
+					text={ 'Добавить поле' }
+					addRemove={ 'add' }
+					click={ itemAdd }
+				/>
+				{ Boolean( index ) && (
+					<ButtonsAddRemoveChild
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/components/buttonsAddRemoveChild/blockButtons.js:38]: I see here duplication of logic. You could refactor it and to create new function that would get `text`, `removeText` and `click` and return `<ButtonsAddRemoveChild ...`

File: src/main/resources/react/src/pages/UserInfo/components/inputProfile/inputGender.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	return (
+		<GenderDiv>
+			<Label>{ 'Пол:' }</Label>
+			<Container>
+				<ChildDiv onClick={ () => props.click( id, 'boy' ) }>
+					<CircleBoy gender={ gender }/>
+					<Span>Мальчик</Span>
+				</ChildDiv>
+				<ChildDiv onClick={ () => props.click( id, 'girl' ) }>
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/components/inputProfile/inputGender.js:78]: Logic duplication. You should extract it into new function that would get `id`, `sex` and return `<ChildDiv ...`

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/myActivity.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+			<TitleBigBlue whatClass={ 'incoming__replies-text' } text={ 'Входящие ответы' }/>
+			<div className={ 'block-of-cards' }>
+				<ProductCard
+					city={ 'Харьков' }
+					text={ `Велосипед ну очень куртой. просто не реально крутой для девочки 5 лет` }
+					picture={ lot3 }
+					inboxMessage={ 12 }
+					avatar={ '' }
+				/>
+				<ProductCard
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/myActivity.js:24]: Duplication of logic. Make new function that will return `<ProductCard ...` and you pass all needed parameters into it.

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/myActivity.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+
+			<TitleBigBlue whatClass={ 'outgoing__replies-text' } text={ 'Исходящие ответы' }/>
+			<div className={ 'block-of-cards' }>
+				<ProductCard isFavorite={ isFavorite }
+					picture={ lot3 } city={ 'Харьков' }
+					text={ `Велосипед ну очень куртой. просто не реально крутой для девочки 5 лет` }
+					inboxMessage={ 3 }
+				/>
+				<ProductCard
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/myActivity.js:68]: Duplication of logic. Would you mind refactor it?

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_favorites/myFavorites.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                <TitleBigBlue whatClass={"myProfile-title"} text={"Избранные объявления"}/>
+                <div className="cards_replies">
+                    <ProductCard
+                      city={ 'Харьков' }
+                      text={ `Велосипед ну очень куртой. просто не реально крутой для девочки 5 лет` }
+                      picture={ "https://static.toiimg.com/photo/72975551.cms" }
+                      isFavorite={ isFavorite }
+                    />
+
+                    <ProductCard
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_favorites/myFavorites.js:31]: Duplication of logic. Refactor this.

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+const MyProfile = () => {
+	const dispatch = useDispatch();
+	const { profile } = useSelector( state => state.profileMe );
+
+	const addInputMe = useCallback( ( e ) => {
+		dispatch( addMeInputValue( { [e.target.name]: e.target.value } ) );
+	}, [ dispatch ] );
+
+	const addInputCity = ( e ) => {
+		console.log( { [e.target.name]: e.target.value } );
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js:32]: We should think about logging all front errors on the server as well. Please remind me about it on the next team meeting.

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+	const addInputPhone = useCallback( ( e, id ) => {
+		dispatch( addPhoneInputValue( { [e.target.name]: e.target.value }, id ) );
+	}, [ dispatch ] );
+
+	const phoneAdd = useCallback(()=> {
+		dispatch(addPhone())
+	},[dispatch])
+
+	const phoneRemove = useCallback(()=> {
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js:43]: Duplication of logic.

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+	const childAdd = useCallback( id => {
+		dispatch( addChild() );
+	}, [ dispatch ] );
+
+	const submitFormAboutMe = ( e ) => {
+		e.preventDefault();
+		dispatch( putUserInfoAsync( profile ) );
+	};
+
+	const submitFormChildren = ( e ) => {
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js:72]: Duplication of logic.

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+				<InputProfile
+					data={ {
+						name: 'lastName',
+						label: 'Фамилия:',
+						value: profile.lastName,
+						type: 'text'
+					} }
+					click={ addInputMe }
+				/>
+				<InputProfile data={ {
```
Comment (rpkyrych) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js:99]: Duplication of logic.

File: src/main/resources/react/src/redux/profile/saga/workers/UserInfoPut.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import { call, put } from 'redux-saga/effects';
+
+import { putUserInfo as putUserInfoAsync } from '../../../../REST/Resources';
+import { fillUserInfoSync } from '../../profileAction';
+import { startFetching, stopFetching } from '../../../ui/action';
+
+export function* userInfoPut(action) {
```
Comment (rpkyrych) [inline src/main/resources/react/src/redux/profile/saga/workers/UserInfoPut.js:7]: I see the same logic into UserInfoGet.js file, so why don't you create only one function for that and additionally pass request type into it?

File: src/main/resources/react/src/redux/ui/action.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import { types } from './types';
+
+export const startFetching = () => ({
+	type: types.START_FETCHING
+});
+export const stopFetching = () => ({
+	type: types.STOP_FETCHING
```
Comment (rpkyrych) [inline src/main/resources/react/src/redux/ui/action.js:7]: What is the sense of such one lines constants?

File: src/main/resources/react/src/redux/ui/reducer.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+		case types.START_FETCHING:
+			return {
+				...state,
+				isFetching: true
+			};
+
+		case types.STOP_FETCHING:
+			return {
+				...state,
+				isFetching: false
```
Comment (rpkyrych) [inline src/main/resources/react/src/redux/ui/reducer.js:18]: Duplication of logic.

File: src/main/resources/react/src/styledTheme/index.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
+export const theme = {
+	colors: {
+		shadow: 'rgba(89, 102, 116, 0.1)',
+		colorPopup: '#FDFDFF',
+		'btn-blue-normal': 'hsl(195, 86%, 50%)', // -> #12b6ed
+		'btn-blue-hover': 'hsl(195, 120%, 50%)', // -> #00bfff
+		'btn-blue-active': 'hsl(205, 100%, 50%)', // -> #0094ff
+		'btn-green-normal': 'hsl(134, 45%, 48%)', // ->  #43b15d
+		'btn-green-hover': 'hsl(134, 45%, 43%)', // -> #3c9f53
+		'btn-green-active': 'hsl(134, 100%, 27%)', // -> #008a20
```
Comment (rpkyrych) [inline src/main/resources/react/src/styledTheme/index.js:10]: Nice example of useful comment! 👍

File: src/main/resources/react/src/styledTheme/index.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+		'colorError': '#ff4c4c', // -> #ff4c4c
+		colorGrey: '#8E8E8E', // -> #8E8E8E
+		'btn-gb-disabled': '#DADADA', // -> #DADADA
+		'right-color-text': '#333333', // -> #333333
+		'black-color-text': '#111111',
+		'bg-profile': '#00C0FF', // -> #00C0FF
```
Comment (rpkyrych) [inline src/main/resources/react/src/styledTheme/index.js:16]: Your comment describes the same meaning of the color, it's completely useless. 👎

File: src/main/resources/react/src/styledTheme/styledFonts.js [hunk 17]
```diff
@@ -0,0 +1,24 @@
+import { createGlobalStyle  } from 'styled-components';
+import PollywogCyr from '../fonts/Pollywog_Cyr.ttf';
+
+export default createGlobalStyle`
+@font-face {
+  @import url('https://fonts.googleapis.com/css?family=Roboto:400,400i,		700&display=swap&subset=cyrillic');
+}
+
+
+	@font-face{@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@800&display=swap');}
```
Comment (rpkyrych) [inline src/main/resources/react/src/styledTheme/styledFonts.js:10]: Formatting problems...

**Review by rpkyrych — APPROVED** [review]
I wrote few comments from mine point of view. Fixing or not it's up to you.

**Review by Jack11M — APPROVED** [review]

**Pull request merged**